bug-02: last station в ответе на POST-запрос по адресу /distance/count не соответствует ОР

AmanEfer opened the pull request:
Исправил багу в классе  DistanceService

AmanEfer committed:
bug-02: last station в ответе на POST-запрос по адресу /distance/count не соответствует ОР
-Исправил багу в классе DistanceService

AmanEfer committed:
bug-02:
- Немного изменил DistanceService, чтобы на переходах поле lastStation было не null
- Добавил метод в Query, который возвращает список last stations по веткам нашего маршрута
-Добавил в AnswerDetailsQuery в подробном отображении last stations для каждой ветки маршрута

AmanEfer committed:
bug-02:
- Пришлось исправлять DistanceService, т.к. сломались тесты
- Подправил AnswerDetailsQuery, потому что так же не проходят тесты. Исправил частично, т.к. не все тесты понимаю как починить

vicria commented on pom.xml:
```diff
@@ -5,9 +5,9 @@
+    <version>1.0.0-SNAPSHOT</version>
     <name>tour-bot-parent-pom</name>
     <packaging>pom</packaging>
-    <version>1.0.0-SNAPSHOT</version>
```
?

AmanEfer replied:
Случайно видимо

vicria commented on subte/subte-microservice/src/main/java/ar/vicria/subte/microservice/services/DistanceService.java:
```diff
@@ -74,9 +76,10 @@ public List<RouteDto> getRoute(StationDto start, StationDto end) {
 
         List<ConnectionDto> transitionsList = new ArrayList<>();
-        // Initialize priority queue with start station and priority 0
+        //Initialize priority queue with start station and priority 0
```
?

AmanEfer replied:
Тоже случайно

vicria commented on telegram/telegram-microservice/src/main/java/ar/vicria/telegram/microservice/services/callbacks/AnswerDetailsQuery.java:
```diff
@@ -79,14 +80,14 @@ private String addTransition(RouteDto send) {
                 .append(firstLine)
-                .append(" ")
+                .append("(")
+                .append(String.format(localized.getLastStation(), lastStations.get(0).getName()))
```
тут в тестах выскакивает NPE

lastStations.get(0) нельзя брать, потому что элементов в листе может и не быть. например пользователь ввел переход между станциями. там нет никакой last station


следовательно надо создать тест между двумя переходами и проверить, что текст для последней станции пути не добавился.

AmanEfer replied:
Готово. 
Логику подправил, тест написал.

AmanEfer committed:
bug-02:
- В AnswerDetailsQuery добавил логику, чтобы не выводилось направление, если пользователь никуда не едет, а просто делает переход
- Написал тест на эту логику
